develop/parachain-devs/runtime-development/FRAME/create-custom-pallet.md

```diff
@@ -0,0 +1,334 @@
+    frame-system = { version = "37.0.0", default-features = false }
+    codec = { version = "3.6.12", default-features = false, package = "parity-scale-codec", features = ["derive"] }
+    scale-info = { version = "2.11.1", default-features = false, features = ["derive"] }
+    sp-runtime = { version = "39.0.0", default-features = false }
+
+    [features]
+    default = ["std"]
+    std = [
+        "frame-support/std",
+        "frame-system/std",
+        "codec/std",
+        "scale-info/std",
+        "sp-runtime/std",
+    ]
+    ```
+
+    !!!note
+        Carefully select the versions of the packages according to your project's specific requirements:
+
+        - When developing for a specific Polkadot SDK runtime, ensure that your pallet's dependency versions match those of the target runtime
+        - If you're creating this pallet within a Polkadot SDK workspace:
+
+            - Define the actual versions in the root `Cargo.toml` file
+            - Use workspace inheritance in your pallet's `Cargo.toml` to maintain consistency across your project
+        - Regularly check for updates to FRAME and Polkadot SDK dependencies to benefit from the latest features, performance improvements, and security patches
+
+        Proper version management is crucial for ensuring compatibility and reducing potential conflicts in your project.
+
+3.  Set up the basic pallet structure
+
+    After setting up the project and configuring dependencies, the next step is to prepare the basic structure of your pallet in the `lib.rs` file. Open the `src/lib.rs` file in your project, remove all existing content and copy and paste the following scaffold code:
+
+    ```rust
+    pub use pallet::*;
+
+    #[frame_support::pallet]
+    pub mod pallet {
+        use frame_support::pallet_prelude::*;
+        use frame_system::pallet_prelude::*;
+
+        #[pallet::pallet]
+        pub struct Pallet<T>(_);
+
+        #[pallet::config]  // snip
+        #[pallet::event]   // snip
+        #[pallet::error]   // snip
+        #[pallet::storage] // snip
+        #[pallet::call]    // snip
+    }
+    ```
+
+    With this scaffold in place, you're ready to start implementing the specific logic and features of your custom pallet. The subsequent sections of this guide will walk you through populating each of these components with the necessary code for your pallet's functionality.
+
+## Pallet Configuration
+
+Every pallet includes a Rust trait called [`Config`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/trait.Config.html){target=\_blank}, which is used to expose configurable options and link your pallet to other parts of the runtime. All types and constants that the pallet depends on must be declared within this trait. These types are defined generically and made concrete when the pallet is instantiated in the `runtime/src/lib.rs` file of your blockchain.
+
+In this step, you'll configure the pallet to emit events. 
```

Comment:
I recommend paraphrasing this sentence

AFAIK each single pallet must configure the following:
-  type RuntimeEvent - so the events emitted by the pallet are accessible to the runtime
- type WeightInfo - that represents the weight that each call of the pallet will have

(btw, unsure if that is the most technical explanation, but it's just to showcase my point here)

So I'd add the explanation of those two types that will be contained in each single pallet, and then, I'd add a sentence that says something like: "You can also add as many types as you need" or something like that.